src/views/ApartmentApp/components/StudentApplication/components/HallChoiceList/index.jsx

```diff
@@ -46,7 +47,20 @@ const HallChoiceList = ({
     };
 
     loadApartmentHalls();
-  }, []);
+
+    if (apartmentChoices?.length >= 5) {
+      setClickCount(5);
+    } else {
+      setClickCount(apartmentChoices?.length);
+    }
+  }, [apartmentChoices]);
+
```

Comment:
```suggestion
    if (apartmentChoices?.length >= 5) {
      setClickCount(5);
    } else {
      setClickCount(apartmentChoices?.length);
    }
  }, [apartmentChoices]);

```

I still do not like the idea of counting clicks. It is not a very robust system. Please use an implementation where we count the number of rendered components